releases/__init__.py

```diff
@@ -598,8 +598,8 @@ def unknown_visit(self, node):
 
 def generate_changelog(app, doctree):
     # Don't scan/mutate documents that don't match the configured document name
-    # (which by default is changelog.rst).
-    if app.env.docname != app.config.releases_document_name:
+    # (which by default is ['changelog.rst', ]).
+    if app.env.docname not in app.config.releases_document_name:
```

Comment:
If you feed a list in as `app.config.release_document_name`, re breaks with the error `TypeError: unhashable type: 'list'`.

Alternately, if you feed `*/changelog.rst` in as `app.config.release_document_name`, re breaks with the error `sre_comstraints.error: nothing to repeat at position 0`.

I feel like I'm missing something here in what you're proposing...